src/model/UserphraseModel.cpp

```diff
@@ -157,9 +157,9 @@ void UserphraseModel::refresh()
     emit refreshCompleted(userphrase_.size());
 }
 
-void UserphraseModel::add(std::shared_ptr<QString> phrase, std::shared_ptr<QString> bopomofo)
+void UserphraseModel::addWrapper(QString &phrase, QString &bopomofo)
```

Comment:
why not just remove this method and move the private `add()` to public?
It's not like the private `add()` is doing anything special that need to hide inside.